src/unix/linux_like/android/mod.rs

```diff
@@ -3649,20 +3643,6 @@ extern "C" {
     pub fn pthread_kill(thread: crate::pthread_t, sig: c_int) -> c_int;
     pub fn sem_unlink(name: *const c_char) -> c_int;
     pub fn daemon(nochdir: c_int, noclose: c_int) -> c_int;
-    pub fn getpwnam_r(
-        name: *const c_char,
-        pwd: *mut passwd,
-        buf: *mut c_char,
-        buflen: size_t,
-        result: *mut *mut passwd,
-    ) -> c_int;
-    pub fn getpwuid_r(
-        uid: crate::uid_t,
-        pwd: *mut passwd,
-        buf: *mut c_char,
-        buflen: size_t,
-        result: *mut *mut passwd,
-    ) -> c_int;
```

Comment:
For these functions and if_nameindex*, I don't mind too much that they get commonized - but could you put any commonization in a separate commit? Since it's distinct from l4re support.

(having orthogonal commits just makes life easier with the backports)